test/server/connection_handler_test.cc

```diff
@@ -604,7 +604,12 @@ TEST_F(ConnectionHandlerTest, NormalRedirect) {
   EXPECT_EQ(1UL, TestUtility::findCounter(stats_store_, "test.downstream_cx_total")->value());
   EXPECT_EQ(1UL, TestUtility::findGauge(stats_store_, "test.downstream_cx_active")->value());
 
-  EXPECT_CALL(*access_log_, log(_, _, _, _)).Times(1);
+  EXPECT_CALL(*access_log_, log(_, _, _, _))
+      .WillOnce(
+          Invoke([&](const Http::RequestHeaderMap*, const Http::ResponseHeaderMap*,
+                     const Http::ResponseTrailerMap*, const StreamInfo::StreamInfo& stream_info) {
+            EXPECT_EQ(alt_address, stream_info.downstreamLocalAddress());
+          }));
```

Comment:
It would be good to exercise this fix in an integration test to fully capture the full implications.

Reply:
Agree, but there is no integration test for original_dst, unfortunately.

Reply:
Could you work with the owners of original_dst to improve test coverage?

Reply:
Who's the owner of original_dst? It's not in CODEOWNERS.

Reply:
I don't know who originally wrote the code (someone in Istio I'm sure). The problem is I'm not sure we can have an integration test that is sandboxed. It requires iptables. If this is possible we should do it.